### ISSUE TITLE
`spago add-from-bower` command

### ISSUE THREAD

**Issue by dariooddenino** [spacchetti/spago#24]
Right now adding a package to the set is a somewhat tedious manual process.
 
Getting inspiration from here: https://github.com/justinwoo/spacchetti/blob/master/scripts/from-bower.pl

It would be cool to be able to do `spacchetti from-bower simple-json` and having it appear as an addition in the packages.dhall file.

**Comment by f-f** [comment]
Maybe actually having the `packages` key in the `spacchetti.dhall` makes sense, as it would cover both this (by specifying a `bower` kind of dependency) and pulling in stuff from git (as detailed in #25, by specifying a `git` kind of dependency)

**Comment by LucianU** [comment]
Are the three features listed above still the desired ones? 

Regarding point 3, what happens if we add the package but its dependencies don't match our package set? If the install will fail, I propose that we don't add the package.

**Comment by f-f** [comment]
@LucianU great questions. I think I now have a slightly different take:

- I changed my mind on the name of this command and I'm now again for the original suggestion, so you'd use it like `spago add-from-bower spork`.
  The reason is that right now 100% of the published packages have a `bower.json` that we can read. Even if in the future you might be able to publish packages without a bowerfile but only with a `spago.dhall`, in that case they'd already be in the package set.
  This also means that "point 3" from above is no longer valid.
- The way this should work is that:
  - We search for the package on the bower registry (eventually with the `purescript-` prefix)
  - The linked repo is fetched in a temporary directory (we should use [`temporary`](http://hackage.haskell.org/package/temporary-1.3/docs/System-IO-Temp.html#v:withSystemTempDirectory) for this, we'll need to pull it anyways when implementing #133) 
  - then we get the `bower.json`, read the dependencies, try to to find them into our local package-set. If we don't find a dependency, we find their repo address (by searching again in the bower registry) and recur
  - This will give us a set of packages we can add to the packages.dhall

Side note: to implement the above we'll need to call the `bower` command (e.g. for `bower search`), and this is fine

**Comment by shubhamkumar13** [comment]
If I had to look where all the commands are created, which file would I look in?

**Comment by f-f** [comment]
@shubhamkumar13 great question! 

So, all commands are specified in [`app/Spago.hs`](https://github.com/spacchetti/spago/blob/master/app/Spago.hs).
The boilerplate for adding a command consists in (I'll use the `spago verify` command as an example):
- adding [a constructor](https://github.com/spacchetti/spago/blob/77fec6dcb765a4bc93220b308b20930176dd04a7/app/Spago.hs#L58) to the `Command` type
- adding [a parser](https://github.com/spacchetti/spago/blob/77fec6dcb765a4bc93220b308b20930176dd04a7/app/Spago.hs#L269) for the command
- [matching](https://github.com/spacchetti/spago/blob/77fec6dcb765a4bc93220b308b20930176dd04a7/app/Spago.hs#L369) the new constructor when you parse the command line

In this last step you then forward the data from the command line to the actual implementation, which in this case should go in the [`Spago.Bower`](https://github.com/spacchetti/spago/blob/master/src/Spago/Bower.hs) module.

You might also want to look into the [`bower-json` package](http://hackage.haskell.org/package/bower-json), which we use to parse the Bower configs

**Comment by shubhamkumar13** [comment]
Sorry for not checking in; so I got a high-level view of how all commands work. So to be clear this command would look through all the packages in dependencies and add it to packages.dhall, right?

It's just parsing bower.json and adding the packages to package.dhall, in layman's terminology.

**Comment by f-f** [comment]
I'll close this as conditions have changed, see [here](https://discourse.purescript.org/t/the-bower-registry-is-no-longer-accepting-package-submissions/1103)
